zingolib/src/lightclient.rs

```diff
@@ -301,12 +303,18 @@ pub struct LightClient {
     // pub(crate) server_uri: Arc<RwLock<Uri>>,
     pub(crate) config: ZingoConfig,
     /// TODO: Add Doc Comment Here!
+    #[cfg(not(feature = "sync"))]
     pub wallet: LightWallet,
+    #[cfg(feature = "sync")]
+    pub wallet: Arc<Mutex<LightWallet>>,
```

Comment:
Mutex instead of RwLock? Is this an intentional disallowing of multiple readers? If so, why?

Reply:
my reasoning is because mutex is more suited to write heavy operations and has less overhead, it will be held by the sync engine for writing 99% of the time. the idea here (as you can see in the main loop of sync.rs) is that the sync engine has a single point after each scan result is processed where it drops the mutex guard for anything outside of sync such as sending or getting the sync status. my theory is that having to wait for the rare case where two read operations are needed between the processing of scan results is more efficient than increasing the time it takes for sync to drop and acquire the guard each cycle. what are your thoughts?

Reply:
I suspect that the extra time to drop and acquire the guard each cycle will be trivial, whereas two processes wanting read access at the same time could in theory make for logistical difficulties...although those could probably be managed by sharing access to the same MutexGuard, if necessary. I don't have a strong opinion on it, I think the tradeoffs are very minor either way.

Reply:
You are probably right, let's wait until the integration is complete and we have benchmarks so we can test the difference and discuss again